### PR TITLE
update room operator override to fit nullsafety requirements

### PR DESCRIPTION
**Pull request by Te-Z**

### CODE REVIEW

File: lib/src/room.dart [hunk 1]
```diff
@@ -2130,7 +2130,7 @@ class Room {
       : setSpaceChild(roomId, via: const []);
 
   @override
-  bool operator ==(dynamic other) => (other is Room && other.id == id);
```
Comment (nqhhdev) [inline lib/src/room.dart:2133]: Can you explain a little bit about this case?

Comment (Te-Z) [reply lib/src/room.dart:2133]: `dynamic` can be null but the operator `==` is not nullable (`bool operator ==`), so the new version of dart require the usage of `Object` here instead to be null safe

Comment (sherlockvn) [reply lib/src/room.dart:2133]: if you could provide a link, it would be great

**Review by nqhhdev — APPROVED** [review]

File: lib/src/room.dart (hunk 1 above)
Comment (Te-Z) [reply lib/src/room.dart:2133]: https://dart.dev/tools/linter-rules/null_check_on_nullable_type_parameter

Comment (Te-Z) [reply lib/src/room.dart:2133]: Also the related error from CI: https://github.com/linagora/twake-on-matrix/actions/runs/9273371007/job/25513989168

**Review by hoangdat — APPROVED** [review]

**Pull request merged**